packages/txservice/src/monitor/monitor.ts

```diff
@@ -0,0 +1,242 @@
+      return result.value as Transaction;
+    } else {
+      throw result.value;
+    }
+  }
+
+  private async getGas(transaction: WriteTransaction): Promise<Gas> {
+    const method = this.getGas.name;
+    // Get gas estimate.
+    let gasLimit: BigNumber;
+    let result = await this.provider.estimateGas(transaction);
+    if (result.isErr()) {
+      if (result.error instanceof TransactionReverted) {
+        // If we get a TransactionReverted error, that means the gas estimate call
+        // indicated our transaction would fail on-chain. The details of the failure will
+        // be included in the error.
+        throw result.error;
+      }
+      this.logger.warn(
+        {
+          method,
+          transaction: transaction,
+          error: result.error,
+        },
+        "Estimate gas failed due to an unexpected error.",
+      );
+      throw result.error;
+    } else {
+      gasLimit = result.value;
+    }
+
+    // Get gas price and create tracker instance.
+    result = await this.provider.getGasPrice();
+    if (result.isErr()) {
+      throw result.error;
+    }
+    const gas = new Gas(result.value, gasLimit);
+    return gas;
+  }
+
+  /**
+   * Get the current nonce value. Should ONLY ever be called within a serialized
+   * queue.
+   *
+   * @remarks
+   * Caller should still be prepared to get the incorrect nonce back. For instance,
+   * if the provider that just handled our sent tx has suddenly gone offline, this
+   * method may give the wrong nonce. This can be solved by making additional calls to
+   * submit the tx.
+   *
+   * @returns A number value for the current nonce.
+   */
+  private async getNonce(): Promise<number> {
+    // Update nonce value to greatest of all nonce values retrieved.
+    const buffer = this.buffer.getLastNonce() ?? -1;
+    const result = await this.provider.getTransactionCount();
+    if (result.isErr()) {
+      throw result.error;
+    }
+    // If transaction buffer above returns null, that indicates the buffer is empty; meaning
+    // we haven't sent any transactions yet, and we'll use this value here.
+    const pending = result.value;
+    // Increment if we got back a value from buffer as we want the value of last tx's nonce + 1.
+    return Math.max(buffer + 1, pending);
+  }
+
+  private async backfillLoop() {
+    // TODO: Make sure this loop is throw-proof
+    while (true) {
+      delay(1000);
+      const pending = this.buffer.pending;
+      // Lazy solution: we only care about a potential hold-up if it could hold anything up.
+      if (pending.length < 2) {
+        continue;
+      }
+      const result = await this.provider.getTransactionCount();
+      if (result.isErr()) {
+        // TODO: log
+      }
+      const currentNonce = result.isOk() ? result.value : -1;
+      const tx: Transaction | undefined = this.buffer.get(currentNonce);
+      if (tx == null) {
+        // BC it could just be the latest nonce and we're currently not doing anything
```

Comment:
is this comment out of date? it seems the case its referencing is handled above (by checking buffer length)